[cloud] ovirt: Add VM lease option to the ovirt_vms module

nijinashok opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Ovirt 4.1 is having a new feature of VM lease for high availability. The PR adds support for configuring this in ovirt_vms module.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
ovirt_vms

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible --version
ansible 2.4.0
  config file = /etc/ansible/ansible.cfg

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_vms.py:
```diff
@@ -130,6 +130,10 @@
             - "If no value is passed, default value is set by oVirt/RHV engine."
+    lease:
+        description:
+            - "Name of the storage domain this virtual machine lease reside on."
```
Please add a note that it's supported since oVirt 4.1.

machacekondra commented on lib/ansible/modules/cloud/ovirt/ovirt_vms.py:
```diff
@@ -619,6 +623,17 @@ def build_entity(self):
             high_availability=otypes.HighAvailability(
                 enabled=self.param('high_availability')
             ) if self.param('high_availability') is not None else None,
+            lease=otypes.StorageDomainLease(
```
```python
lease=otypes.StorageDomainLease(
  storage_domain=otypes.StorageDomain(
    id=get_id_by_name(
      service=self._connection.system_service().storage_domains_service(),
      name=self.param('lease')
    ),
  ),
) if self.param('lease') is not None else None,
```

machacekondra reviewed (changes requested):
Looks good , thank you. I just have two minor comments.

nijinashok committed:
[cloud] ovirt: Add VM lease option to the ovirt_vms module

Ovirt 4.1 is having new feature of VM lease for high availability.
The commit add support for configuring this in ovirt_vms module.

nijinashok commented:
Thank you @machacekondra. Changed as per the last comment.

machacekondra commented:
Thanks! @mwperina @ryansb shipit

mwperina reviewed (approved):
Looks good to me, thanks! 👍

mwperina commented:
Ryan, could you please merge it?

@ryansb
